### ISSUE TITLE
Local IAM user is not persisted in NAS gateway mode

### ISSUE THREAD

**Issue by vdm** [minio/minio#13739]
## Expected Behavior
I expect a local IAM user created in NAS gateway mode to persist along with the rest of the minio config.

## Current Behavior
A local IAM user seems to disappear after as few as 30 to as many as 300 seconds. I have not seen a pattern in the timing other than this observed range.

## Steps to Reproduce (for bugs)

Start in NAS gateway mode:
```
$ MINIO_ROOT_USER=admin MINIO_ROOT_PASSWORD=admin123 ./minio gateway nas shared/ --console-address :33157
```
Add user and set user policy:
```
$ (set +o history; ./mc admin user add myminio user1 user1234); ./mc admin policy set myminio readonly user=user1; (while true; do echo $(date) $(./mc admin user info myminio user1); sleep 10; done) 
Added user `user1` successfully.
Policy `readonly` is set on user `user1`
Tue Nov 23 09:03:08 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:03:18 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:03:28 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:03:38 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:03:48 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:03:59 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:04:09 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:04:19 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:04:29 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:04:39 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:04:49 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:04:59 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:05:09 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:05:19 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:05:29 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:05:39 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:05:49 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:05:59 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:06:09 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:06:19 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:06:29 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:06:39 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:06:49 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:06:59 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:07:09 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:07:19 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:07:30 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:07:40 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
Tue Nov 23 09:07:50 PM GMT 2021 AccessKey: user1 Status: enabled PolicyName: readonly MemberOf:
mc: <ERROR> Unable to get user info. The specified user does not exist. (Specified user does not exist).
Tue Nov 23 09:08:00 PM GMT 2021
```

## Context
I would like to use the minio NAS gateway to get clients adapted ahead of deploying a distributed minio cluster.

## Regression
TBD

## Your Environment

* Version used:
```
$ ./minio -version
minio version RELEASE.2021-11-09T03-21-45Z
```
* Operating System and version:
```
$ uname -a
Linux fedora 5.14.17-301.fc35.x86_64 #1 SMP Mon Nov 8 13:57:43 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by vdm** [comment]
This looks like the same problem: https://www.truenas.com/community/threads/minio-s3-loses-configs-on-restart.94199/

**Comment by harshavardhana** [comment]
Yes this is fixed here #13694
